pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -623,7 +623,12 @@ func (a *sqlInstanceAdapter) Update(ctx context.Context, updateOp *directbase.Up
 			desiredEdition = *a.desired.Spec.Settings.Edition
 		}
 
-		if desiredEdition != a.actual.Settings.Edition {
+		actualEdition := "ENTERPRISE" // Default value
```

Comment:
I think we need to determine the right behavior for MySQL 8.4 which defaults to ENTERPRISE_PLUS.